### PR TITLE
Fix test failure TestKnnByteVectorQuery#testTimeLimitingKnnCollectorManager

### PR DESCRIPTION
**Pull request by easyice**
`TopKnnCollectorManager#newCollector` will return `MultiLeafKnnCollector` instance if the index has multiple segments

```
org.apache.lucene.search.TestKnnByteVectorQuery > testTimeLimitingKnnCollectorManager FAILED
    java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([A0DCCB76BFBF940C:8E1E7418F5C62B46]:0)
        at org.junit.Assert.fail(Assert.java:87)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.junit.Assert.assertTrue(Assert.java:53)
        at org.apache.lucene.search.BaseKnnVectorQueryTestCase.testTimeLimitingKnnCollectorManager(BaseKnnVectorQueryTestCase.java:787)
        at org.apache.lucene.search.TestKnnByteVectorQuery.testTimeLimitingKnnCollectorManager(TestKnnByteVectorQuery.java:30)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
```

```
gradlew test --tests TestKnnByteVectorQuery.testTimeLimitingKnnCollectorManager -Dtests.seed=A0DCCB76BFBF940C -Dtests.nightly=true
```

### CODE REVIEW

File: lucene/core/src/test/org/apache/lucene/search/BaseKnnVectorQueryTestCase.java [hunk 1]
```diff
@@ -784,7 +785,9 @@ public void testTimeLimitingKnnCollectorManager() throws IOException {
           noTimeoutManager.newCollector(Integer.MAX_VALUE, searcher.leafContexts.getFirst());
 
       // Check that a normal collector is created without timeout
-      assertTrue(noTimeoutCollector instanceof TopKnnCollector);
+      assertTrue(
```
Comment (dweiss) [inline lucene/core/src/test/org/apache/lucene/search/BaseKnnVectorQueryTestCase.java:788]: Thanks for trying to fix this. I think this type of test is problematic in that it restricts the implementations that are, in theory, an open set... I would change the implementation returned inside TimeLimitingKnnCollectorManager to a package-private class, then check for that using assertFalse(noTimeoutCollector instanceof TimeLimitingKnnCollectorManager.TimeoutingKnnCollector)?

Comment (easyice) [reply lucene/core/src/test/org/apache/lucene/search/BaseKnnVectorQueryTestCase.java:788]: Thanks for you suggestions! you are right, I like the idea!

**Review by dweiss — APPROVED** [review]

**Review by kaivalnp — APPROVED** [review]
Thanks for fixing this!

File: lucene/core/src/java/org/apache/lucene/search/TimeLimitingKnnCollectorManager.java [hunk 2]
```diff
@@ -42,54 +42,62 @@ public KnnCollector newCollector(int visitedLimit, LeafReaderContext context) th
...
-                ? TotalHits.Relation.GREATER_THAN_OR_EQUAL_TO
-                : docs.totalHits.relation;
-
-        return new TopDocs(new TotalHits(docs.totalHits.value, relation), docs.scoreDocs);
-      }
-    };
+    return new TimeoutingKnnCollector(collector);
+  }
+
+  class TimeoutingKnnCollector implements KnnCollector {
```
[the remark this answers is not in the document]
Comment (easyice) [reply lucene/core/src/java/org/apache/lucene/search/TimeLimitingKnnCollectorManager.java:48]: Thank you @kaivalnp , it looks better, fixed.

**Pull request merged**